### ISSUE TITLE
Bug: Warning callback-based function handlers caused by @tracer.captureLambdaHandler()

### ISSUE THREAD

**Issue by ncino-esselman** [aws-powertools/powertools-lambda-typescript#4306]
### Expected Behavior

I would expect to be able to add `@tracer.captureLambdaHandler()` to my async lambda handler and not recieve the following warning

>  AWS Lambda plans to remove support for callback-based function handlers starting with Node.js 24. You will need to update this function to use an async handler to use Node.js 24 or later. For more information and to provide feedback on this change, see https://github.com/aws/aws-lambda-nodejs-runtime-interface-client/issues/137. To disable this warning, set the AWS_LAMBDA_NODEJS_DISABLE_CALLBACK_WARNING environment variable.

### Current Behavior

We recently started getting:

>  AWS Lambda plans to remove support for callback-based function handlers starting with Node.js 24. You will need to update this function to use an async handler to use Node.js 24 or later. For more information and to provide feedback on this change, see https://github.com/aws/aws-lambda-nodejs-runtime-interface-client/issues/137. To disable this warning, set the AWS_LAMBDA_NODEJS_DISABLE_CALLBACK_WARNING environment variable.

We are not using callback based lambda handler definitions and after removing the `@tracer.captureLambdaHandler()` from our handler function the warning goes away. 

### Code snippet

```
  @tracer.captureLambdaHandler()
  @logger.injectLambdaContext()
  public async handler(event: any, context: Context): Promise<any> {
    await this.preRun(event, context);
    return await this.main(event, context);
  }



export const handler = new Handler();
export const main = handler.handler.bind(handler);
```

### Steps to Reproduce

1. Create a simple async handler
2. Deploy and run and see no warning in logs
3. Add `@tracer.captureLambdaHandler()` to handler() definition
4. deploy and run and see the warning

### Possible Solution

```
public captureLambdaHandler(
  options?: CaptureLambdaHandlerOptions
): HandlerMethodDecorator {
  return (_target, _propertyKey, descriptor) => {
    const originalMethod = descriptor.value!;
    const tracerRef = this;
    
    // Check if the original method expects a callback parameter
    const originalMethodLength = originalMethod.length;
    const expectsCallback = originalMethodLength >= 3;
    
    // Create the wrapper function with the appropriate signature
    if (expectsCallback) {
      // Original method expects callback - create wrapper with callback parameter
      descriptor.value = function (this: Handler, event, context, callback) {
        // ... existing logic with callback parameter
      } as SyncHandler<Handler> | AsyncHandler<Handler>;
    } else {
      // Original method doesn't expect callback - create wrapper WITHOUT callback parameter
      descriptor.value = function (this: Handler, event, context) {
        // ... existing logic without callback parameter
      } as SyncHandler<Handler> | AsyncHandler<Handler>;
    }

    return descriptor;
  };
}
```
ref https://github.com/aws-powertools/powertools-lambda-typescript/blob/081a514b6e1cb81513080352e5dbcadc89ac0574/packages/tracer/src/Tracer.ts#L422-L473

### Powertools for AWS Lambda (TypeScript) version

latest

### AWS Lambda function runtime

22.x

### Packaging format used

npm

### Execution logs

```Shell
INIT_START Runtime Version: nodejs:20.v70	Runtime Version ARN: arn:aws:lambda:us-east-1::runtime:c11827d1307a63ca6bbb996dbffba66628e1328ed4714070f4b891884a323e0b
2025-08-11T15:27:00.879Z	undefined	WARN	AWS Lambda plans to remove support for callback-based function handlers starting with Node.js 24. You will need to update this function to use an async handler to use Node.js 24 or later. For more information and to provide feedback on this change, see https://github.com/aws/aws-lambda-nodejs-runtime-interface-client/issues/137. To disable this warning, set the AWS_LAMBDA_NODEJS_DISABLE_CALLBACK_WARNING environment variable.
```

**Comment by dreamorosi** [comment]
Hi @ncino-esselman, thanks for opening this report. 

Could you please share your `tsconfig.json` configuration file?

**Comment by ncino-esselman** [comment]
Sure
```
{
    "compilerOptions": {
      "target": "ES2020",
      "module": "commonjs",
      "lib": ["es2020"],
      "declaration": true,
      "strict": true,
      "noImplicitAny": true,
      "strictNullChecks": true,
      "noImplicitThis": true,
      "alwaysStrict": true,
      "noUnusedLocals": false,
      "noUnusedParameters": false,
      "noImplicitReturns": true,
      "noFallthroughCasesInSwitch": false,
      "inlineSourceMap": true,
      "inlineSources": true,
      "experimentalDecorators": true,
      "strictPropertyInitialization": false,
      "typeRoots": ["./node_modules/@types"],
      "outDir": "dist"
    },
    "exclude": ["cdk.out", "dist"]
  }
```

**Comment by dreamorosi** [comment]
Thank you for sharing the config.

I tested the Tracer with a setup like yours and I am able to reproduce the issue. We'll try investigating if we can provide a backwards compatible fix to maintain the type of handler you set.

**Comment by ncino-esselman** [comment]
Thank you let me know if I can contribute!

**Comment by dreamorosi** [comment]
Apologies, I just saw your message as I opened the PR - the proposed solution involves avoiding destructuring the `args` of the decorated method so that they're passed as they appear, which in my tests silences the warning.

[We are investigating](https://github.com/aws-powertools/powertools-lambda-typescript/pull/4312#issuecomment-3179693490) if the issue affects other Powertools for AWS packages - if this is the case and you're still interested we'd love to have your contribution there.

**Comment by dreamorosi** [comment]
Will keep this open until we have an answer about the comment above.

**Comment by ncino-esselman** [comment]
Sounds good thank you!
